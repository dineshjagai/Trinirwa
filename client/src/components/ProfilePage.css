.all_boxes {
    display: flex;
    flex-direction: row;
    justify-content: flex-start;
}

.all_boxes > div {
    margin: 10px;
}

.left-boxes > div {
    margin: 10px;
}

.left-boxes {
    display: flex;
    flex-direction: column;
}

.left-friends {
    
}