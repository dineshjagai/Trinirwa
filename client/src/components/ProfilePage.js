import axios from "axios";
import Profile from './ProfilePicture';
import NavBar from './navBar';
import Displayer from './Displayer';
import Followers from './Followers';
import React, {useState, useEffect} from 'react';
import TweetDisplayer from './tweetDisplayer.js';
import { BrowserRouter as Router, Route } from "react-router-dom";


export default function ProfilePage(props) {
  const words = (window.location.href).split('/');
  const id = words[words.length - 1];
  const url = `/profile/${id}`;
    const[info, setData] = useState([{}]);
    const[followers, setFollowers] = useState({});
    const[interests, setInterests] = useState(new Set());
    const [made_query, setMadeQuery] = useState(false);
    const[tweets, setTweets] = useState([{}]);
    const [childKey, setChildKey] = useState(0);
  
  const fetchTweets = ()=>{
      const getUrl = '/profile/tweet/2';
      axios({
        method: 'GET',
        url: getUrl,
      }).then((result)=>{
          setTweets(result.data);
      });
}
   
  useEffect(()=>{
    axios({
      method: 'get',
      url: url
    }).then((result)=>{
        setData(result.data.data[0]);
        setFollowers(result.data.followers);
        let interests = new Set((result.data.interests).map(obj=>obj.interest))
        setInterests(interests);
        setMadeQuery(true);
        fetchTweets();
     });
  }, []);
 
  const addInterest  = (newInterest) => {
    setInterests([...interests, newInterest])
    setChildKey(Math.floor(Math.random() * 1000000000))
    const addUrl = '/profile/interest/2';
    axios({
        method: 'POST',
        url: addUrl,
        data: {
            interest: newInterest,
        }
    });
}


const deleteInterest = (interes)=>{
        let temp = interests;
        temp.delete(interes);
        setInterests(temp);
        setChildKey(Math.floor(Math.random() * 1000000000))
        const delUrl= `/profile/delete/interest/${2}`;//props.id;
        axios({
            method: 'DELETE',
            url: delUrl,
            data: {
                interest: interes,
            }
        });
    }

  return made_query ? (
    <div className="profile">
      
       <NavBar/>
       <div>
          <Profile picture= {info}/>
          <br></br>
          <br></br>
       </div>
        <div className = "center"> 
        
           <Displayer key={childKey} id = {3} interests={interests} addInterest = {addInterest} deleteInterest={deleteInterest}/>
            <br></br>
          <Followers input_followers={followers}/> 
        </div>
        <div id = "center"className="center">
           <TweetDisplayer tweets={tweets}/> 
        </div>
    </div>
  ) : <div></div>;
}


export default ProfilePage;