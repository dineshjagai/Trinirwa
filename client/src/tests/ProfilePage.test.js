import React from 'react';
import renderer from 'react-test-renderer';
import ProfilePage from '../components/ProfilePage';

test('Test ProfilePage snapshot', () => {
  const component = renderer.create(<ProfilePage 
    profile_picture="logo512.png" 
    username="arnaud"
    />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
