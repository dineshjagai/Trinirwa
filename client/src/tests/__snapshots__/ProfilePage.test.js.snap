// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test ProfilePage snapshot 1`] = `<div />`;

exports[`Test Regisration snapshot 1`] = `<div />`;
